include/nds/fifocommon.h

```diff
@@ -335,20 +335,18 @@ int fifoGetDatamsg(int channel, int buffersize, u8 * destbuffer);
 	\param channel the channel to check.
 
 */
-//---------------------------------------------------------------------------------
-static inline void fifoWaitValue32(int channel) {
-//---------------------------------------------------------------------------------
-
-	while(!fifoCheckValue32(channel)) {
-#ifdef ARM9
-		cothread_yield_irq(IRQ_FIFO_NOT_EMPTY);
-#else
-		swiIntrWait(1,IRQ_FIFO_NOT_EMPTY);
-#endif
-	}
 
+static inline void fifoWaitValue32(int channel)
+{
+	while (!fifoCheckValue32(channel))
+		swiIntrWait(1, IRQ_FIFO_NOT_EMPTY);
 }
 
+static inline void fifoWaitValueAsync32(int channel)
```

Comment:
It might be good to add equivalent Wait/WaitAsync functions for Address and Datamsg too.

Reply:
That's fair, but I think that can be added outside of this PR, it's a bit complicated as it is now.

Reply:
OK